pkg/envoy/ads/server.go

```diff
@@ -53,7 +53,7 @@ func (s *Server) Start(ctx context.Context, cancel context.CancelFunc, port int,
 	}
 
 	xds_discovery.RegisterAggregatedDiscoveryServiceServer(grpcServer, s)
-	go utils.GrpcServe(ctx, grpcServer, lis, cancel, ServerType)
+	go utils.GrpcServe(ctx, grpcServer, lis, cancel, ServerType, nil)
```

Comment:
You don't want to be passing a nil channel, writing to it will block.

Reply:
Agreed. Let's use this channel and set `s.ready = true` if it's successful